### DISCUSSION TITLE
Object organization: nested vs flat

### DISCUSSION THREAD

**Discussion by nbenn** [BristolMyersSquibb/blockr.core#2] [Ideas]
In v1 we opted for a "nested" structure: the workspace _contains_ stack, with each _containing_ blocks with each _containing_ fields. This could also be arranged in a "flat" structure, where the workspace contains stack and block objects and association is tracked via "links" (i.e. IDs).

- We might want to keep with the nesting of fields as components of blocks
- If stacks and blocks are merely "associated", whose responsibility is keeping track of this information? Does the stack maintain the list of its blocks? Or does each block know which stack(s) it belongs to?

**Comment by DivadNojnarg** [comment]
> We might want to keep with the nesting of fields as components of blocks

Yes, especially if we want to have dynamic field numbers, it would make to many modules at the same level. Flattening at the stack level could already be a great simplification in term of module communication, especially to handle connecting stack together.
